addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.js

```diff
@@ -116,7 +116,7 @@ class DiscussSidebar extends Component {
                 thread.isPinned &&
                 thread.model === 'mail.channel'
             )
-            .sort((c1, c2) => c1.displayName < c2.displayName ? -1 : 1);
+            .sort((c1, c2) => c1.displayName && c1.displayName.toLowerCase() < c2.displayName && c2.displayName.toLowerCase() ? -1 : 1);
```

Comment:
While we are at it, please expand and fix the sort for other cases too:
- make explicit what happens when one channel has a displayName and the other not? I guess the one with name should be above?
- then have the lowercase condition
- then if both have the same name, id should be used as last resort, to ensure the order is deterministic

Reply:
@seb-odoo 
1) I think, there is no case if any of the channels have a name but not displayName, 
refer, https://github.com/odoo/odoo/blob/14.0/addons/mail/static/src/models/thread/thread.js#L1063

So just need to check with the case insensitiveness and the Id if the name is same.

Reply:
@seb-odoo ok Thanks.
Now we have changed the implementation way. just have a look.